lint.js

```diff
@@ -2,19 +2,23 @@
 
 const fs = require("fs").promises;
 
+const isString = obj =>
+  Object.prototype.toString.call(obj) === "[object String]";
+
 // Lint by normalizing specs.json and comparing it to the original,
 // fixing it in place if |fix| is true.
 async function lint(fix = false) {
   const specsBuffer = await fs.readFile("./specs.json");
-  let specs = JSON.parse(specsBuffer);
-
-  const fixed = specs.map(spec => {
-    const url = new URL(spec.url).toString();
-    return { url };
-  });
-  fixed.sort((a, b) => {
-    a.url.localeCompare(b.url);
-  });
+  const specs = JSON.parse(specsBuffer);
+
+  const sorted = specs
+    .map(spec => isString(spec) ? { url: spec } : spec)
+    .map(spec => Object.assign({}, spec, { url: new URL(spec.url).toString() }));
```

Comment:
This allows for arbitrary additional properties, most likely in the form of typos if we add more properties. I'd prefer this code to normalize everything that is known and drop anything else. Since currently no additional properties are allowed, just require all entries to be a string.

Reply:
I was assuming that we would in any case add some kind of JSON schema to validate the structure before we pass it to the linter. I'll add some code logic to that effect.

Reply:
I started out using ajv myself, but ripped it out since the schema was so trivial. Now when either a string or object are allowed it's less trivial, so I'm happy with using it.

I'll have a few days off for the lunar new year, but will review this when I'm back next week!

Reply:
I just added a JSON Schema and schema validation. The schema is a bit dumb for now since it allows an object with a URL, but then that's useless as long as we don't have additional properties, especially since the linting process will convert objects to URL strings. At least we'll have the right skeleton and logic in place for when we add properties.